src/components/opacityscale/attributes.js

```diff
@@ -0,0 +1,59 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var palettes = require('./scales.js').scales;
+var paletteStr = Object.keys(palettes);
+
+/**
+ * Make opacityscale attribute declarations for
+ *
+ * - opacityscale,
+ *
+ * @param {string} context (dflt: '', i.e. from trace root):
+ *     the container this is in ('', *marker*, *marker.line* etc)
+ *
+ * @param {object} opts:
+ *   - opacityscaleDflt {string}:
+ *     overrides the opacityscale dflt
+ *
+ *   - editTypeOverride {boolean} (dflt: ''):
+ *     most of these attributes already require a recalc, but the ones that do not
+ *     have editType *style* or *plot* unless you override (presumably with *calc*)
+ *
+ * @return {object}
+ */
+module.exports = function opacityScaleAttrs(context, opts) {
```

Comment:
We don't need a new `components/` module just for this thing. I can't imagine we'll need `opacityscale` for traces other than `volume` at any point.

Could you merge these new files into one and place it in  `traces/volume/opacityscale.js`.